src/custom/state/orders/updater.ts

```diff
@@ -69,227 +52,58 @@ function _computeFulfilledSummary({
-    if (lastCheckedBlock + 1 > lastBlockNumber) return
-
-    const getPastEvents = async () => {
-      console.log('EventUpdater::getLogs', {
-        fromBlock: lastCheckedBlock + 1,
-        toBlock: lastBlockNumber,
-        address: contractAddress,
-        topics: eventTopics
-      })
-
-      const logs = await getLogsRetry({
-        fromBlock: lastCheckedBlock + 1,
-        toBlock: lastBlockNumber,
-        address: contractAddress,
-        topics: eventTopics
-      })
-
-      const block2DateMap = await buildBlock2DateMap(library, logs)
-
-      // Filter out orders that should not trigger a pop-up
-      const pendingLogsAndOrders = logs.reduce<[Log, Order][]>((acc, log) => {
-        const { orderUid: id } = decodeTradeEvent(log)
-
-        console.log(`EventUpdater::Detected Trade event for order ${id} of token in block`, log.blockNumber)
-
-        const orderFromStore = findOrderById(id)
-        if (!orderFromStore || orderFromStore.status !== 'pending') {
-          console.log(`EventUpdater::Order ${id} not pending or not on local storage, ignoring it`)
-          return acc
-        }
-
-        acc.push([log, orderFromStore])
-
-        return acc
-      }, [])
-
-      const ordersBatchData: OrderLogPopupMixData[] = await Promise.all(
-        pendingLogsAndOrders.map(async ([log, orderFromStore]) => {
-          const id = orderFromStore.id
-
-          // We've found the orderId in the Trade event
-          // But the backend may not be completely updated yet, e.g
-          // the frontend is ahead of the backend in regards to data freshness
-          // TODO: temporary! change to a better solution
-          // https://github.com/gnosis/gp-swap-ui/issues/213
-          const orderFromApi = await delay(DEFAULT_ORDER_DELAY, getOrder(chainId, id))
-
-          // using order from store and api compute summary
-          const summary = _computeFulfilledSummary({ orderFromApi, orderFromStore })
-
-          return {
-            id,
-            fulfillmentTime: block2DateMap[log.blockHash].toISOString(),
-            transactionHash: log.transactionHash,
-            summary
-          }
-        })
+  const updateOrders = useCallback(
+    async (pending: Order[], chainId: ChainId) => {
+      // Iterate over pending orders fetching operator order data, async
+      // Returns a null when the order isn't finalized/not found
+      const unfilteredOrdersData: (OrderLogPopupMixData | null)[] = await Promise.all(
+        pending.map(async orderFromStore => fetchOrderPopupData(orderFromStore, chainId))
       )
 
-      // SET lastCheckedBlock = lastBlockNumber
-      // AND fulfill orders
-      // ordersBatchData can be empty
+      // Additional step filtering out `null` entries, due to lack of async reduce
+      // Type guard is required to tell TS what's the final type
+      const ordersData = unfilteredOrdersData.filter((data): data is OrderLogPopupMixData => data !== null)
```

Comment:
nice!

Reply:
My suggestion of using unfilteredOrdersData.filter(Boolean)

![image](https://user-images.githubusercontent.com/2352112/113189090-1a8c1780-925b-11eb-8679-fa409b5864bc.png)

Reply:
TS doesn't like that:
![screenshot_2021-03-31_11-12-15](https://user-images.githubusercontent.com/43217/113191040-edffdd80-9211-11eb-90e1-748723bb83ee.png)